### PR TITLE
fix(api): do not add default tags if there already some tags setted

### LINKED ISSUE #2532: fix : delete a tag in the sidebar
tag is not deleted

cc @jithinjk

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

1. Description
1. Related issues
close #2532 
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/workflow.go [hunk 1]
```diff
@@ -213,8 +212,8 @@ func (api *API) putWorkflowHandler() Handler {
 			return sdk.WrapError(errHr, "putWorkflowHandler> HookRegistration")
 		}
 
-		if wf.Root.Context.DefaultPayload != nil || (wf.Root.Context != nil && wf.Root.Context.Application != nil && wf.Root.Context.Application.RepositoryFullname != "") {
-			wf.Metadata = getUpdatedMetadata(wf.Metadata)
+		if defaultTags, ok := wf.Metadata["default_tags"]; wf.Root.IsLinkedToRepo() && (!ok || defaultTags == "") {
+			wf.Metadata["default_tags"] = "git.branch,git.author"
```
Comment (fsamin) [inline engine/api/workflow.go]: log.go:149: [ERROR]    [PANIC_RECOVERY] Panic occurred on PUT:/Test_putWorkflowHandler/project/3zmjcqyqlu/workflows/Name, recover assignment to entry in nil map

			/go/src/github.com/ovh/cds/engine/api/workflow.go:216 +0x776

**Comment by sguiheux** [comment]
You  had to start from WasC branch no ?

**Comment by bnjjj** [comment]
@sguiheux no :(

**Comment by bnjjj** [comment]
check the #2555

**Pull request closed without merging**